server/src/test/java/org/opensearch/cluster/structure/RoutingIteratorTests.java

```diff
@@ -860,4 +860,77 @@ public void testWeightedRoutingShardStateWithDifferentWeights() {
             terminate(threadPool);
         }
     }
+
+    /**
+     * Test to validate that simple weighted shard routing with seed return same shard routing on each call
+     */
+    public void testActiveInitializingShardsSimpleWeightedIt() {
+        TestThreadPool threadPool = null;
+        try {
+            Settings.Builder settings = Settings.builder()
+                .put("cluster.routing.allocation.node_concurrent_recoveries", 10)
+                .put("cluster.routing.allocation.awareness.attributes", "zone");
+            AllocationService strategy = createAllocationService(settings.build());
+
+            Metadata metadata = Metadata.builder()
+                .put(IndexMetadata.builder("test").settings(settings(Version.CURRENT)).numberOfShards(1).numberOfReplicas(2))
+                .build();
+
+            RoutingTable routingTable = RoutingTable.builder().addAsNew(metadata.index("test")).build();
+
+            ClusterState clusterState = ClusterState.builder(ClusterName.CLUSTER_NAME_SETTING.getDefault(Settings.EMPTY))
+                .metadata(metadata)
+                .routingTable(routingTable)
+                .build();
+
+            threadPool = new TestThreadPool("testThatOnlyNodesSupport");
```

Comment:
Copy/paste error on "testThatOnlyNodesSupport"? Also can probably move this assignment up and skip assigning `null` to `threadPool`.